includes/blocks/class-sensei-page-blocks.php

```diff
@@ -18,6 +18,44 @@ class Sensei_Page_Blocks extends Sensei_Blocks_Initializer {
 	 */
 	public function __construct() {
 		parent::__construct( [ 'page' ] );
+
+		add_filter( 'render_block_core/post-featured-image', [ $this, 'add_a_badge'], 10, 3 );
+		add_filter( 'render_block', array( $this, 'add_course_featured_badge' ), 11, 3 );
+	}
+	public function add_a_badge( string $block_content, array $block, WP_Block $instance ) {
+		// Add featured course badge to a featured image block.
+		if ( empty( $block_content ) || 'featured' !== get_post_meta( $instance->context['postId'], '_course_featured', true ) ) {
+			return $block_content;
+		}
+		return '<div class="featured-image-wrapper"><div class="featured-badge" style="display: block">Featured</div>' . $block_content . '</div>';
+	}
+
+	/**
+	 * A function to add a course featured badge to the course list.
+	 *
+	 * @access public
+	 * @since $$next-version$$
+	 *
+	 * @param string $block_content This is block content.
+	 * @param object $block_parent This is block parent.
+	 *
+	 * @return string $block_content block content.
+	 */
+	public function add_course_featured_badge( $block_content, $block_parent, WP_Block $instance  ): string
+	{
+		if('sensei-lms/course-categories' !== $block_parent['blockName']) {
```

Comment:
This statement could use whitespace as the other `if` statement does below.

Reply:
a679a7877e675c77f3b369c7905e4e8249552f81